libs/application-generic/src/utils/sanitize-control-values.ts

```diff
@@ -222,37 +237,63 @@ export function dashboardSanitizeControlValues(
   controlValues: Record<string, unknown>,
   stepType: StepTypeEnum | unknown,
 ): (Record<string, unknown> & { skip?: Record<string, unknown> }) | null {
-  if (!controlValues) {
-    return null;
-  }
-  let normalizedValues: Record<string, unknown>;
-  switch (stepType) {
-    case StepTypeEnum.IN_APP:
-      normalizedValues = sanitizeInApp(controlValues as InAppControlType);
-      break;
-    case StepTypeEnum.EMAIL:
-      normalizedValues = sanitizeEmail(controlValues as EmailControlType);
-      break;
-    case StepTypeEnum.SMS:
-      normalizedValues = sanitizeSms(controlValues as SmsControlType);
-      break;
-    case StepTypeEnum.PUSH:
-      normalizedValues = sanitizePush(controlValues as PushControlType);
-      break;
-    case StepTypeEnum.CHAT:
-      normalizedValues = sanitizeChat(controlValues as ChatControlType);
-      break;
-    case StepTypeEnum.DIGEST:
-      normalizedValues = sanitizeDigest(
-        controlValues as DigestControlSchemaType,
-      );
-      break;
-    case StepTypeEnum.DELAY:
-      normalizedValues = sanitizeDelay(controlValues as DelayControlType);
-      break;
-    default:
-      normalizedValues = filterNullishValues(controlValues);
+  try {
+    if (!controlValues) {
+      return null;
+    }
+
+    console.log('controlValues 333222 ', controlValues);
+    let normalizedValues: Record<string, unknown>;
+    switch (stepType) {
+      case StepTypeEnum.IN_APP:
+        normalizedValues = sanitizeInApp(controlValues as InAppControlType);
+        break;
+      case StepTypeEnum.EMAIL:
+        normalizedValues = sanitizeEmail(controlValues as EmailControlType);
+        break;
+      case StepTypeEnum.SMS:
+        normalizedValues = sanitizeSms(controlValues as SmsControlType);
+        break;
+      case StepTypeEnum.PUSH:
+        normalizedValues = sanitizePush(controlValues as PushControlType);
+        break;
+      case StepTypeEnum.CHAT:
+        normalizedValues = sanitizeChat(controlValues as ChatControlType);
+        break;
+      case StepTypeEnum.DIGEST:
+        normalizedValues = sanitizeDigest(
+          controlValues as DigestControlSchemaType,
+        );
+        break;
+      case StepTypeEnum.DELAY:
+        normalizedValues = sanitizeDelay(controlValues as DelayControlType);
+        break;
+      default:
+        normalizedValues = filterNullishValues(controlValues);
+    }
+
+    console.log('normalizedValues 333222 ', normalizedValues);
+
+    return normalizedValues;
+  } catch (error) {
+    console.error('Error sanitizing control values', error);
```

Comment:
i made this sanitization more safe, i think it will be a good idea to pass here a logger and print Errors as well.

Reply:
Why not logger now?

Reply:
done